### PR TITLE
Ease first setup, introduce entrypoint providers for vendor & phpunit & symfony

### PR DESCRIPTION
**Pull request by janedbal**

### CODE REVIEW

File: src/Provider/PhpUnitEntrypointProvider.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            yield (string) $tag->value;
+        }
+    }
+
+    /**
+     * @return iterable<string>
+     */
+    private function getDataProvidersFromAttributes(ReflectionMethod $method): iterable
+    {
+        if (PHP_VERSION_ID < 80_000) {
```
Comment (JanTvrdik) [inline src/Provider/PhpUnitEntrypointProvider.php:116]:
```suggestion
        if (PHP_VERSION_ID < 8_00_00) {
```

File: src/Provider/PhpUnitEntrypointProvider.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            return;
+        }
+
+        foreach ($method->getAttributes('PHPUnit\Framework\Attributes\DataProvider') as $providerAttributeReflection) {
```
Comment (JanTvrdik) [inline src/Provider/PhpUnitEntrypointProvider.php:120]: Why not use `::class`?

File: src/Provider/SymfonyEntrypointProvider.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        if (!$this->enabled) {
+            return false;
+        }
+
+        $methodName = $method->getName();
+
+        return $method->getDeclaringClass()->implementsInterface('Symfony\Component\EventDispatcher\EventSubscriberInterface')
+            || (PHP_VERSION_ID >= 80_000 && $method->getAttributes('Symfony\Contracts\Service\Attribute\Required') !== [])
+            || (PHP_VERSION_ID >= 80_000 && $method->getAttributes('Symfony\Component\Routing\Attribute\Route') !== [])
+            || $this->isProbablySymfonyListener($methodName);
```
Comment (JanTvrdik) [inline src/Provider/SymfonyEntrypointProvider.php:29]: Missing support for `AsEventListener` attribute

File: src/Provider/PhpUnitEntrypointProvider.php (hunk 2 above)
Comment (janedbal) [reply src/Provider/PhpUnitEntrypointProvider.php:120]: Does not really matter, would cause CDA reports.

**Pull request merged**